tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -0,0 +1,273 @@
+                                                    nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed IOs before soft-failure.")
+
+        LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                            deploy=self.deply_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        iam_output = Queue()
+        io_output = Queue()
+        event = threading.Event()
+
+        LOGGER.info("Step 2: Starting independent background threads for IOs on already created "
+                    "user and IAM user creation/deletion.")
+        LOGGER.info("Step 2.1: Start IAM user creation/deletion in background during control pod "
+                    "soft-failure.")
+
+        args = {'user_crud': True, 'num_users': self.num_users,
+                'output': iam_output, 'del_users_dict': self.del_users, 'header': self.header}
+        thr_iam = threading.Thread(target=self.ha_obj.iam_bucket_cruds,
+                                   args=(event, ), kwargs=args)
+        thr_iam.daemon = True  # Daemonize thread
+        self.test_io_prefix = 'test-45498-io'
+        args = {'s3userinfo': list(self.s3_clean.values())[0], 'log_prefix': self.test_io_prefix,
+                'nclients': 1, 'nsamples': 8, 'output': io_output, 'setup_s3bench': False}
+        LOGGER.info("Step 2.2: Start IOs in background during control pod soft-failure.")
+        io_thread = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        io_thread.daemon = True  # Daemonize thread
+        io_thread.start()
+        thr_iam.start()
+        LOGGER.info("Step 2: Started independent background threads for IOs on already created "
+                    "user and IAM user creation/deletion.")
+
+        LOGGER.info("Step 3: Perform soft-failure of %s control pods in loop and check cluster "
+                    "has failures", self.repl_num - 1)
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        event.set()
+        for pod in pod_list:
+            self.node_master_list[0].kill_process_in_container(pod_name=pod,
+                                                               container_name=const.CORTX_CSM_POD,
+                                                               process_name=const.CSM_AGENT_PRC,
+                                                               safe_kill=True)
+            LOGGER.info("Step 3.1: Check cluster has failures.")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
```

Comment:
Why are you expecting false here?
As you have not added any delay in restart, container would restart immediately and cluster will be in good state only.

Reply:
I have observed cluster status is not coming clean for few secs even though restart of container.
Test where failing for the moment that's why kept false, It's working as expected and all test logs are taken. Hence, keeping as it is as this is what getting observed